### PR TITLE
Instantly resolve onEmpty if the queue is empty

### LINKED ISSUE #12: Have queue.onEmpty() instantly resolve when queue.size === 0
I'm currently using the following pattern to simulate said behavior:

```js
if (queue.size > 0) await queue.onEmpty();
doSomethingAfterQueueIsFinished();
```

Which could be simplified to

```js
await queue.onEmpty();
doSomethingAfterQueueIsFinished();
```

if `queue.onEmpty()` would return a resolved promise when `queue.size === 0`. Currently, the promise from `onEmpty` will never resolve, unless another item is added to the queue, and finishes.

Would love to know other peeps' thoughts.

### PR DESCRIPTION
**Pull request by edorivai**
Resolves #12

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -113,6 +113,11 @@ class PQueue {
 	}
 
 	onEmpty() {
+		// Instantly resolve if the queue is empty
+		if (this.queue.size === 0) {
```
Comment (floatdrop) [inline index.js]: Here should be `&& this.pending === 0`, fixed this in master.

Comment (floatdrop) [reply index.js]: Or should it be?

Comment (edorivai) [reply index.js]: Ah, cool. Wasn't sure if you guys were okay with this already... Should this also be fixed in `this._resolveEmpty`?

Comment (edorivai) [reply index.js]: I see two options:

1. Promise from onEmpty resolves when `size === 0 && pending === 0`
2. Two separate methods; onEmpty and onIdle (not sure about naming, but you get the gist)

Comment (floatdrop) [reply index.js]: @edorivai I'm not sure, but is there a case, when we only need `size === 0` for queue?

@sindresorhus It was long time ago, but is this behaviour of `.onEmpty` intentional?

If we are about to bump major version, this should be reconsidered.

Comment (floatdrop) [reply index.js]: I would go for second option, because it matches with [async.queue](https://github.com/async-js/async#queue) naming.

Comment (floatdrop) [reply index.js]: But from `Advanced example` it looks like first option is anticipated – and I guess this is what most people expect from queue. If we add some work to it – `onEmpty` should mean, that all work is done.

Comment (edorivai) [reply index.js]: Shall we open a separate issue for the `pending === 0` functionality?

Comment (floatdrop) [reply index.js]: @edorivai sure.

**Pull request merged**